lib/Email/MIME/Encode.pm

```diff
@@ -16,6 +17,10 @@ sub maybe_mime_encode_header {
     $header = lc $header;
 
     my $header_length = length($header) + length(": ");
+
+    return $val->as_mime_string($charset, $header_length)
```

Comment:
I've changed the code to use named arguments and avoid the whole question of order.